docs/containers-auth.json.5.md

```diff
@@ -11,7 +11,7 @@ on Windows and macOS, at `$HOME/.config/containers/auth.json`.
 
 When searching for the credential for a registry, the following files will be read in sequence until the valid credential is found:
 first reading the primary (read/write) file, or the explicit override using an option of the calling application.
-If credentials are not present, search in `${XDG\_CONFIG\_HOME}/containers/auth.json`, `$HOME/.docker/config.json`, `$HOME/.dockercfg`.
+If credentials are not present, search in `${XDG_CONFIG_HOME}/auth.json`, `$HOME/.docker/config.json`, `$HOME/.dockercfg`.
```

Comment:
If I recall correctly, the backslashes are needed for some kind of obscure md functionality.

I'm sure about this change.  As noted in the [documentation]{https://docs.podman.io/en/latest/markdown/podman.1.html?highlight=XDG_CONFIG_HOME#xdg-config-home}, we pull information from $XDG_CONFIG_HOME if it is defined, otherwise, we pull it from $HOME/.config/containers.  What gets a bit confusing is by default, $XDG_CONFIG_HOME defaults to $HOME/.config

Given that, I think having the `containers` in there is correct.  @svdHero first of all, TY for the PR.  Have you seen this fail for you?  Was that the impetus for this change?  i.e. have I been reviewing too many things and I'm off-base here?

Reply:
> If I recall correctly, the backslashes are needed for some kind of obscure md functionality.

I guess they were necessary in a not-code-marked version, where underscores are a syntax for italic type.

I agree that they are incorrect here, and should be removed.

---

> According to [documentation](https://docs.podman.io/en/latest/markdown/podman.1.html?highlight=XDG_CONFIG_HOME#xdg-config-home), the default value for `XDG_CONFIG_HOME` is `$HOME/.config/containers`.

That documentation is a bit confusing, and could be improved. It is correct but it could easily be read to mean that `XDG_CONFIG_HOME defaults to `$HOME/.config/containers, which is not the case.

The **containers-auth.json**(5) text is correct as is, comparing it with the `pkg/docker/config/config.go` implementation.

---

> How can I verify that it actually has been used when doing a podman login harbor.mycompany.com?

`podman login --verbose` outputs where it wrote the credentials.

As for checking whether they have been _found and used_, debug-level logs (`podman --log-level=debug`, `skopeo --debug`) show where credentials were found. They don’t list files that were checked with no match, that would be interesting to add.

---

> Maybe it would be good to also add this to the `auth.json` docs, too? I can do that in another commit while also reverting my false correction.

The documentation should continue to refer to `XDG_CONFIG_HOME`, which is what _actually happens_, but e.g. a parenthetical “(usually `~/.config/containers/auth.json)” might be quite useful.

Reply:
> As for checking whether they have been _found and used_, debug-level logs (`podman --log-level=debug`, `skopeo --debug`) show where credentials were found. They don’t list files that were checked with no match, that would be interesting to add.

https://github.com/containers/image/pull/1445 .